### PR TITLE
OPB-02: Raise error on oversized chain id

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

Fixes OPB-02.

This is a draft, POC for how I'd do this in signer.go. Two other locations still need fixing which I'll do in a subsequent PR once this is approved.

**Tests**

No new tests are added as it's unclear how this would actually be reachable, however it was decided to include it as "future proofing".

### CODE REVIEW

File: op-node/p2p/signer.go [hunk 1]
```diff
@@ -82,13 +88,14 @@ type SignerSetup interface {
 
 // LoadSignerSetup loads a configuration for a Signer to be set up later
 func LoadSignerSetup(ctx *cli.Context) (SignerSetup, error) {
-	key := ctx.GlobalString(flags.SequencerP2PKeyFlag.Name)
```
Comment (maurelian) [inline op-node/p2p/signer.go:85]: not sure how the changes below here were made. Gotta tame my IDE formatter it looks like.

File: op-node/p2p/signer.go [hunk 2]
```diff
@@ -23,19 +23,23 @@ type Signer interface {
...
-func SigningHash(domain [32]byte, chainID *big.Int, payloadBytes []byte) common.Hash {
+func SigningHash(domain [32]byte, chainID *big.Int, payloadBytes []byte) (common.Hash, error) {
 	var msgInput [32 + 32 + 32]byte
 	// domain: first 32 bytes
 	copy(msgInput[:32], domain[:])
 	// chain_id: second 32 bytes
+	// check that chain_id is not too large
+	if chainID.BitLen() > 256 {
+		return common.Hash{}, errors.New("chain_id is too large")
+	}
```
Comment (maurelian) [inline op-node/p2p/signer.go:34]: This is the basic change which addresses the audit recommendation. All other changes are downstream of it.

**Review by protolambda — APPROVED** [review]
LGTM. Technically the `FillBytes` usages are all safe; chain ID is loaded exclusively from local configuration. And the other usages from OPB-2 are all through other safe code paths (deposit fields are from test only code-paths, and basefee size is checked by the json decoder). Still nice to catch any potential issues when the code is used wrong or called in unexpected ways however. Nice fix. The other OPB-02 fixes can also go in a separate PR.

**Comment by maurelian** [comment]
@trianglesphere seems to all be working now

**Review by trianglesphere — APPROVED** [review]

**Review by tynes — APPROVED** [review]

**Comment by mslipper** [comment]
CircleCI errored out and didn't mark completed jobs as done, so they hung. Restarted the jobs to get this through.

**Pull request merged**